Client/src/Unified/Services/Submitter/Service.cs

```diff
@@ -94,22 +99,71 @@ public class Service : AbstractClientService, ISubmitterService
     : base(properties,
            loggerFactory)
   {
+    var timeOutSending = properties.TimeTriggerBuffer ?? TimeSpan.FromSeconds(1);
+
+    var maxTasksPerBuffer = properties.MaxTasksPerBuffer;
+
+    semaphoreSlim_ = new SemaphoreSlim(properties.MaxConcurrentBuffer * maxTasksPerBuffer);
+
+    queue_ = Channel.CreateUnbounded<string>(new UnboundedChannelOptions());
```

Comment:
Do you really need an unbounded queue? I thought the maximum number of elements was known at this point.

Reply:
Good point. Fix it